### DISCUSSION TITLE
I need help with my power strip

### DISCUSSION THREAD

**Discussion by koko009** [arendst/Tasmota#16527] [Q&A]
Good evening or good morning
First, I am not a programmer, nor am I a professional, but I am an amateur and have somewhat experience

I have this power strip and it is from LG and it is a smart wifi, but I cannot run it on its program because it is Korean and it requires a Korean membership with a Korean phone number
So I thought until I arrived that I will install esp8266 inside it with flashing tasmota, but I have a problem
All I have found so far is
1\ I set the program to generic 18
2\ I set the output number GPIO13 to relay i
3\ I connected the two wire of the gnd esp8266 to the wire of the gnd power strip
4\ I connected GPIO13 to KEY 4 on power strip

Now when I click on Toggle it turns off every two clicks and turns on every two clicks
I can't yet how to fix it

All I want is to run LGUplus or run esp8266 while keeping the buttons

<a href="https://ibb.co/3BdkPyw"><img src="https://i.ibb.co/m6TSPCm/20220914-231545.jpg" alt="20220914-231545" border="0" /></a>
<a href="https://ibb.co/GQD7DdZ"><img src="https://i.ibb.co/7tDvDS3/20220914-231615.jpg" alt="20220914-231615" border="0" /></a>
<a href="https://ibb.co/87fdYxf"><img src="https://i.ibb.co/x7BfFgB/20220914-231642.jpg" alt="20220914-231642" border="0" /></a>
<a href="https://ibb.co/M1x0BjB"><img src="https://i.ibb.co/jwPXJxJ/20220914-231737.jpg" alt="20220914-231737" border="0" /></a>
<a href="https://ibb.co/T2wrrxm"><img src="https://i.ibb.co/tKH33T8/20220914-233413.jpg" alt="20220914-233413" border="0" /></a>

**Comment by sfromis** [comment]
The MCU chip in the next-to-last image has the Realtek logo, meaning that it is not Espressif ESP8266, but a Tasmota-incompatible chip. Not feasible to even replace, without "major surgery".

Not clear what you're trying to do with the dev board, but hooking up extra chips needs extra care with matching electrics and power to work.

**Reply by koko009** [reply]
I understand that the Realtek chip is not compatible with Tazmota. But I want to install a esp 8266 with it, but I don't know how to set it. The problem now is that I want to press the Toggle button to give a short pulse in order to turn it off, then press again to give a short pulse to turn it on. I hope you understand me

**Reply by sfromis** [reply]
Now you say "short pulse", and Tasmota has a feature for that; you can set a PulseTime to have each "power on" only be a pulse of the selected length.

**Reply by koko009** [reply]
I dont know how to setup this 
But step by step 
Iam not devloper

**Reply by sfromis** [reply]
Using basic Tasmota commands like [PulseTime](https://tasmota.github.io/docs/Commands/#pulsetime) is not a question of being a "developer", just getting to know Tasmota. When you use the command to configure a PulseTime, it will be persistent across boots.

**Reply by koko009** [reply]
Can you explain more how to use on gpio 13
I want click on toggle one to make short second

**Reply by koko009** [reply]
Sorry, but more clarification. I want when I press the toggle, shorts occur for second only not more between gpio. and GND

**Reply by barbudor** [reply]
Read the docs for 'pulsetime' command
That will allow you to get that short pulse on relay output

Alternatively, you may try 'Relay_b' instead of 'Relay' which is meant for bistable relays that change state by pulse

**Reply by sfromis** [reply]
The MCU chip cannot itself "short" two pins. What it can do is to output a weak signal near 3.3V or 0V, which with common ground *may* work to influence the other system, as long as you *first* make sure that the systems have compatible electrics and power. Or you may need some external component, like a relay or optocoupler.

**Reply by koko009** [reply]
First of all, sorry, I have not been able to test pulse time until now because I do not know how to write its formula, but what I have come up with so far is. I select the output mode relay_bi, then go to the console and type power on and nothing happens at all. I wrote power off, his pulse went out
What I understand now is that when pressing toggle ON it does not work, but when pressing toggle OFF it works properly

**Reply by koko009** [reply]
I set PulseTime to 1 
Now Every time I press it, it actually gives a pulse, and this is what I want, but when I press the button so that it becomes ON after a second, it becomes OFF automatically, which I do not want.I want a pulse to be ON and steady until I press again and a pulse is given to turn OFF.

**Reply by sfromis** [reply]
A pulse is by definition only ON for a brief period, the opposite of "steady". Don't worry about the state you see in Tasmota, this is only about what Tasmota does, not about what the other device may have done as a result. Tasmota cannot "know" if the strip is powered on or off by a pulse.

**Reply by koko009** [reply]
But knowing the status of the device, whether it is on or off, this is important, and now whether it is on or off, it will give an OFF result on the screen.I want to press it to give me the screen OFF, then another press to give me the screen ON.
OR........
Why when I type the command power on it does not give any result and when I type power off it gives me a result?

**Reply by sfromis** [reply]
I'm assuming that you want Tasmota to act as a momentary push button toggling the strip on vs off for every push. Imagine having to turn on or off by hand, without being able to know if it currently is on or off.

What you "want" may not be what you are actually enabling. If Tasmota has to detect whether the strip happens to be on or off, you need some sort of sensor input, when Tasmota cannot command the strip to be on, without being sure that it is currently off, or vice versa.

If you are *only* changing strip state through Tasmota, and nothing else, you could have Tasmota "simulate" a state by counting times it sends a pulse, but with no guarantee of this being the real state.

**Reply by koko009** [reply]
Iam sorry I did not understand you, but in this video he explains the problem and what I want is that when you press the button once, it turns ON and gives a pulse, then press again and it gives OFF and also gives a pulse
<div style="width:100%;height:0px;position:relative;padding-bottom:178.218%;"><iframe src="https://streamable.com/e/stoget" frameborder="0" width="100%" height="100%" allowfullscreen style="width:100%;height:100%;position:absolute;left:0px;top:0px;overflow:hidden;"></iframe></div>
https://streamable.com/stoget

**Reply by sfromis** [reply]
Stop repeating what you *want* to happen, and start looking at what's possible with the build you have.

**Reply by koko009** [reply]
Sorry I was repeating myself because I wasn't sure you understood what I wanted to do. Now, am I not able to do what I want to do? Sorry if I misunderstood what you said.

**Reply by sfromis** [reply]
If I understood your build correctly, Tasmota cannot "know" the real current state of the strip.
> act as a momentary push button toggling the strip on vs off for every push

But still:
> If you are only changing strip state through Tasmota, and nothing else, you could have Tasmota "simulate" a state by counting times it sends a pulse, but with no guarantee of this being the real state.

**Reply by koko009** [reply]
First, I apologize for taking up a lot of your time
Secondly, I would like to communicate with you in private so that I can explain to you more accurately and understand you more accurately as well, but excuse me now, I will explain the matter in all its details so that you understand what I want exactly
I have in the power strip four buttons for each output. When I press the button once, the output is turned on, and when I press again, the output is turned off.
When tracing the path of these buttons, he found them with the names KEY1 / KEY2 / KEY3 / KEY4
When I make shorts between GND and KEY1, OUTPUT 1 is turned ON
And when I make shorts again, the OUTPUT is OFF
All I want is for the esp8266 to make these shorts by controlling through the browser {toggle}

**Reply by sfromis** [reply]
I already told you that ESP8266 cannot make shorts, and now you are just repeating what you want, without even considering what your hardware setup is capable of.

I'm not open to taking it private, a public forum is better when anyone can jump in if they have something to add. And anyone (including me) can refrain from commenting if not having anything further to add.

**Reply by koko009** [reply]
Thank you for clarifying all these things, but let me clarify something for you and at the same time understand whether there is a solution to it or not Now I added the output mode to relay_bi and I also made an addition to pulsetime 1 and now it works and makes short shorts with each time I press the toggle But I have a small problem and I hope to solve it, which is when I press the button, the output is turned on and it writes to me on the screen ON, but after about 5 seconds it is written on the screen OFF, but the OUTPUT remains working All I want is that when I press the button and it says ON, it stays like this without changing automatically and writes to me OFF.

**Reply by barbudor** [reply]
DON'T MIX Relay_B and PulseTime
Use one or the other

**Reply by sfromis** [reply]
You just keep restating what you want, and ignore my explanations of why Tasmota cannot work that way. And I did suggest a potential strategy, with the downside that it cannot be reliable.

**Reply by koko009** [reply]
I am not completely ignoring your explanations, I am just not a professional and I am afraid that I may have misunderstood you or could not convey the information to you correctly, so I told you what I did and what I reached. And now I explained to you what I did, and you say that there is a possible strategy, is this a solution? What if this is a solution? Please excuse me, I am trying very hard to understand what you mean and what you are saying so as not to overburden you or stress you with me.

**Reply by koko009** [reply]
> DON'T MIX Relay_B and PulseTime
> Use one or the other

If you choose this option, a esp8266 does not make shorts, only sends a signal, but I want to send shorts [ push button ]

**Reply by techman83** [reply]
@koko009 - you have an electronics problem to solve, not a software one. You can either use a relay to replace the action of the button, or get an understanding of how a mosfet works. You *might* get away with sending a "low" if you can share grounds, and the circuits share a ground (as mentioned in this thread already). But you are well into YMMV territory there.

You have two problems to solve:
- "Pressing" the button
- Tracking power state

Personally I'd find an off the shelf powerboard with an ESP in it, admittedly a lot are being replaced with non-esp based chips. But some use a drop in replacement of the TYWE2S that could be replaced (likely a bit out of reach for the early stages of learning).

The simplest you could achieve is electrically decouple the problem with a relay. Have it short the pins of the switch and track the state of the switch elsewhere. It's a common method of integrating with Garage doors that have a rolling code remote.

Once you understand the electrical side of your problem, it's probably going to be a little easier to understand you want out of the software side.
